### PR TITLE
Fix ptr.annotation translation for pointers on non-struct types

### PR DESCRIPTION
**Pull request by mlychkov**
Signed-off-by: Mikhail Lychkov <mikhail.lychkov@intel.com>

### CODE REVIEW

**Comment by mlychkov** [comment]
Tagging @AlexeySachkov @AlexeySotkin @AGindinson @MrSidims

File: test/transcoding/fpga_lsu_optimized.ll [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+;   int output_data[2];
+;   for (size_t i = 0; i < 2; i++) {
+;     output_data[i] = -1;
+;   }
+;
+;   int input_data[2];
+;   for (size_t i = 0; i < 2; i++) {
+;     input_data[i] = i + 1;
+;   }
+;
```
Comment (AlexeySachkov) [inline test/transcoding/fpga_lsu_optimized.ll:17]:
```suggestion
```

I think there is no need to put host code in there as it anyway is not emitted to LLVM IR. The same applies to code that validates results after kernel enqueue

Comment (mlychkov) [reply test/transcoding/fpga_lsu_optimized.ll]: Thank you for a quick feedback. Removed.

**Comment by MrSidims** [comment]
Minor: I suggest to add 'vendor name_' prefix for vendor-specific tests.

**Comment by mlychkov** [comment]
> Minor: I suggest to add 'vendor name_' prefix for vendor-specific tests.

Yes, I agree, good suggestion. Thank you.

**Pull request merged**